### PR TITLE
`[dynamic_layouts]` Added the example template

### PR DESCRIPTION
**Pull request by snat-s**
Added the example template to fill later with examples.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like the checks are rather unhappy. Can you take a look?

File: packages/dynamic_layouts/example/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# This file configures the analyzer, which statically analyzes Dart code to
```
Comment (goderbauer) [inline packages/dynamic_layouts/example/analysis_options.yaml:1]: Remove this file? There is a global analyze options for this repository.

**Review by goderbauer — APPROVED** [review]
LGTM once checks are happy.

**Comment by cbracken** [comment]
For the `SetNextFrameCallback` issue my guess is that CI is running on `stable` branch, but the example was created with `flutter create` on `master` branch. 

The `SetNextFrameCallback` was recently added but is currently only available on `master`.

**Comment by Piinks** [comment]
> For the SetNextFrameCallback issue my guess is that CI is running on stable branch, but the example was created with flutter create on master branch.
>
> The SetNextFrameCallback was recently added but is currently only available on master.

@cbracken does that mean this example app should be recreated with the stable version of Flutter in order to land? It sounds like it will not pass CI in this current state until those changes roll to stable

**Comment by Piinks** [comment]
@snat-s said the `example` app was created on the stable branch of Flutter.. I am not sure why it is still failing due to the windows change on master. The other failures are from the analyzer and license check.

File: packages/dynamic_layouts/example/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,115 @@
+import 'package:flutter/material.dart';
```
Comment (Piinks) [inline packages/dynamic_layouts/example/lib/main.dart:1]: Just double checking my own sanity (fever 😅 ), this is not the sample app, it's the newly created app while debugging the CI issue?

**Pull request merged**